### PR TITLE
feat(utils): extract `isNotTokenOfTypeWithConditions` out of `ast-utils`' `predicates`

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
This is the negative version of what I did in #3977

### CODE REVIEW

File: packages/utils/src/ast-utils/helpers.ts [hunk 1]
```diff
@@ -56,3 +56,24 @@ export const isTokenOfTypeWithConditions = <
...
+    conditions: Conditions,
+  ): ((
+    token: TSESTree.Token | null | undefined,
+  ) => token is Exclude<
+    TSESTree.Token,
+    TSESTree.Token & { type: TokenType } & Conditions
+  >) =>
+  (
+    token,
+  ): token is Exclude<
```
Comment (MichaelDeBoey) [inline packages/utils/src/ast-utils/helpers.ts:75]: Unfortunately, linting obliges me to add this type a second time 😢

**Review by JoshuaKGoldberg — APPROVED** [review]
❤️‍🔥

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Ah, actually, codecov is right - could you make sure these are unit tested please?

At least `isNotOptionalChainPunctuator` is duplicated in user code.

**Comment by MichaelDeBoey** [comment]
@JoshuaKGoldberg The other helper functions also don't have any tests, so I think this is good to go as is?

**Comment by JoshuaKGoldberg** [comment]
Hmm, I suppose so. It would be nice to have them be unit tested directly but I guess if none of them are we can have that as a followup issue. 🤷

**Comment by JoshuaKGoldberg** [comment]
This one I definitely cannot merge without an adult's help:

Screenshot of a failing GitHub status check blocking merge

**Review by JoshuaKGoldberg — APPROVED** [review]
✅

**Comment by JoshuaKGoldberg** [comment]
This one I definitely cannot merge without an adult's help:

Screenshot of failing expected checks blocking merge

**Pull request merged**